PythonVisualizations/QuadTree.py

```diff
@@ -0,0 +1,227 @@
+import random
+from tkinter import *
+from VisualizationApp import *
+
+
+class Node(object):
+    def __init__(self, x, y, d):
+        self.x = x
+        self.y = y
+        self.data = [d] # might be more than one data at this coord 
+        self.dataObjects = [] #keeps track of text objects
+        self.NE = self.SE = self.SW = self.NW = None
+        self.countSpaces = 0
+        #used for drawing intersecting lines
+        self.horizontal_line = None
+        self.vertical_line = None
+
+class PointQuadtree(VisualizationApp):
+    MAX_ARG_WIDTH = 3
+    LINE_COLOR = 'SteelBlue4'
+    CIRCLE_DIMEN = 8
+     
+    def __init__(self, maxArgWidth = MAX_ARG_WIDTH, title="Point Quad Tree", **kwargs):
+        super().__init__(title=title, maxArgWidth = maxArgWidth, **kwargs)
+        self.__root = None
+        self.buttons = self.makeButtons()
+        self.points = []
+        self.nodes = []
+        self.lines = []
+        self.COUNTER = 1
+        self.direction = None #used in the draw lines function
+        self.parent = None    #allows for creating coords of intersecting lines with recursion
+        self.showTree = False
+        self.rootOutline = None
+        self.border = self.canvas.create_rectangle((2, 2, 801, 401),width = 2, outline = self.LINE_COLOR)
```

Comment:
I think changing the canvas background color would be a better way to show the extent of the canvas than using a canvas rectangle.  That allows the edges to move as the window expands and contracts and helps with the Resize bug.

This can be done with a command like `self.canvas['background'] = 'cyan'`.  The color should probably be a named constant of the class.